File: Problemset/1614-Maximum Nesting Depth of the Parentheses/Count.rb
# @param {String} s
# @return {Integer}
def max_depth(s)
  left_count = 0
  ret = 0

  s.chars.each do |ch|
    if ch == '('
      left_count += 1
      ret = left_count if left_count > ret
    elsif ch == ')'
      left_count -= 1
    end
  end

  ret
end

File: Problemset/1614-Maximum Nesting Depth of the Parentheses/Count.rs
impl Solution {
    pub fn max_depth(s: String) -> i32 {
        let mut left_count = 0;
        let mut ret = 0;

        for ch in s.chars() {
            match ch {
                '(' => {
                    left_count += 1;
                    ret = ret.max(left_count);
                }
                ')' => left_count -= 1,
                _ => (),
            }
        }

        ret
    }
}
